src/hooks/thirdPartyFacet/useCleanSarcophagus.ts

```diff
@@ -37,5 +32,7 @@ export function useCleanSarcophagus(sarcoId: string, paymentAddress: string | un
     ...config,
   });
 
-  return { clean: write, isLoading, isCleaning, isSuccess };
+  console.log('isError', isError);
+
+  return { clean: write, isCleaning, isSuccess, isError, mayFail };
```

Comment:
what is the purpose of returning `isError` and `mayFail`? aren't they the same value?

Reply:
`mayFail` comes from `usePrepareContractWrite`'s `isError` (line 12 above), indicating that the call will fail if the `write` function is actually called. 

`isError` comes from `useContractWrite`, and will *only* be set if `write` is called (in this case aliased to `clean`) and reverts.

It's a pretty cool way of preemptively preventing users from making the contract call if it'll fail anyway, which is what I was trying to learn about here, and left it in for visibility in case it'll be needed elsewhere. 

Although in this case, the button wouldn't show at all if the sarco isn't cleanable, so it shouldn't be possible to call `clean` if `mayFail` is true.